source/ui/PGM/PGMPersistentView.cpp

```diff
@@ -0,0 +1,178 @@
+
+    bpm11ID1valueX_ = new QLabel("   um");
+    bpm11ID1valueX_->setFont(QFont("Lucida Grande", 10, QFont::Bold));
+
+    bpm11ID1valueY_ = new QLabel("   um");
+    bpm11ID1valueY_->setFont(QFont("Lucida Grande", 10, QFont::Bold));
+
+    bpm11ID2valueX_ = new QLabel("   um");
+    bpm11ID2valueX_->setFont(QFont("Lucida Grande", 10, QFont::Bold));
+
+    bpm11ID2valueY_ = new QLabel("   um");
+    bpm11ID2valueY_->setFont(QFont("Lucida Grande", 10, QFont::Bold));
+
+    bpm10IDLabel_ = new QLabel("10ID DnStr");
+    bpm10IDLabel_->setAlignment(Qt::AlignLeft);
+    bpm11ID1Label_ = new QLabel("11ID #1");
+    bpm11ID1Label_->setAlignment(Qt::AlignLeft);
+    bpm11ID2Label_ = new QLabel("11ID #2");
+    bpm11ID2Label_->setAlignment(Qt::AlignLeft);
+
+    bpmXValueLabel_ = new QLabel("X: ");
+    bpmXValueLabel_->setAlignment(Qt::AlignCenter);
+    bpmYValueLabel_ = new QLabel("Y: ");
+    bpmYValueLabel_->setAlignment(Qt::AlignCenter);
+
+    QGridLayout *bpmLayout = new QGridLayout();
+
+    bpmLayout->addWidget(bpm10IDLabel_,   0, 1, 1, 1);
+    bpmLayout->addWidget(bpm11ID1Label_,  0, 2, 1, 1);
+    bpmLayout->addWidget(bpm11ID2Label_,  0, 3, 1, 1);
+    bpmLayout->addWidget(bpmXValueLabel_, 1, 0, 1, 1);
+    bpmLayout->addWidget(bpmYValueLabel_, 2, 0, 1, 1);
+    bpmLayout->addWidget(bpm10IDvalueX_,  1, 1, 1, 1);
+    bpmLayout->addWidget(bpm10IDvalueY_,  2, 1, 1, 1);
+    bpmLayout->addWidget(bpm11ID1valueX_, 1, 2, 1, 1);
+    bpmLayout->addWidget(bpm11ID1valueY_, 2, 2, 1, 1);
+    bpmLayout->addWidget(bpm11ID2valueX_, 1, 3, 1, 1);
+    bpmLayout->addWidget(bpm11ID2valueY_, 2, 3, 1, 1);
+
+    bpmLayout->setColumnMinimumWidth(1, 100);
+    bpmLayout->setColumnMinimumWidth(2, 100);
+    bpmLayout->setColumnMinimumWidth(3, 100);
+
+    bpmBox_ = new QGroupBox("Beam Position Monitor");
+    bpmBox_->setLayout(bpmLayout);
+}
+
+void PGMPersistentView::setupBPMConnections(){
+    // BPM 10ID-x connections
+    connect(PGMBeamline::pgm()->bpm10IDxControl(), SIGNAL(onValueChanged(QString)), bpm10IDvalueX_, SLOT(setText(QString)));
+    connect(PGMBeamline::pgm()->bpm10IDxControl(), SIGNAL(isValidStateChanged(bool)), this, SLOT(onBPM10IDxStatusChanged(bool)));
+    // BPM 10ID-x connections
+    connect(PGMBeamline::pgm()->bpm10IDyControl(), SIGNAL(onValueChanged(QString)), bpm10IDvalueY_, SLOT(setText(QString)));
+    connect(PGMBeamline::pgm()->bpm10IDyControl(), SIGNAL(isValidStateChanged(bool)), this, SLOT(onBPM10IDyStatusChanged(bool)));
+    // BPM 10ID-x connections
+    connect(PGMBeamline::pgm()->bpm11ID1xControl(), SIGNAL(onValueChanged(QString)), bpm11ID1valueX_, SLOT(setText(QString)));
+    connect(PGMBeamline::pgm()->bpm11ID1xControl(), SIGNAL(isValidStateChanged(bool)), this, SLOT(onBPM11ID1xStatusChanged(bool)));
+    // BPM 10ID-x connections
+    connect(PGMBeamline::pgm()->bpm11ID1yControl(), SIGNAL(onValueChanged(QString)), bpm11ID1valueY_, SLOT(setText(QString)));
```

Comment:
Does this actually work?  Controls only emit a valueChanged(double) signal.  I'm not sure where your `onValueChanged(QString)` is coming from.